src/GithubActionsToolHelper.ts

```diff
@@ -0,0 +1,89 @@
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+import * as exec from '@actions/exec';
+
+export class GitHubActionsToolHelper {
+
+    public getBuildId(): string {
+        return process.env['GITHUB_RUN_ID'] || '';
+    }
+
+    public getBuildNumber(): string {
+        return process.env['GITHUB_RUN_NUMBER'] || '';
+    }
+
+    public writeInfo(message: string): void {
+        core.info(message);
+    }
+
+    public writeError(message: string): void {
+        core.error(message);
+    }
+
+    public writeWarning(message: string): void {
+        core.warning(message);
+    }
+
+    public writeDebug(message: string): void {
+        core.debug(message);
+    }
+
+    public async exec(commandLine: string, args?: string[], inputOptions?: Buffer): Promise<{ exitCode: number, stdout: string, stderr: string }> {
+        try{
+        let stdout = '';
+        let stderr = '';
+
+        const options: exec.ExecOptions = {
+            listeners: {
+                stdout: (data: Buffer) => {
+                    stdout += data.toString();
+                },
+                stderr: (data: Buffer) => {
+                    stderr += data.toString();
+                },
+            },
+            input: inputOptions
+        };
+
+        let exitCode =  await exec.exec(commandLine, args, options);
+        return new Promise((resolve, reject) => {
+            let executionResult = {
+              exitCode: exitCode,
+              stdout: stdout,
+              stderr: stderr
+            }
+            resolve(executionResult);
+          });
+        }catch(err){
+            throw err;
+        }
+    }
+
+    public getInput(name: string, options?: core.InputOptions): string {
```

Comment:
Rather than this function accepting an `options?: core.InputOptions` argument, which is specific to the `@actions/core` library, we should change the argument to be `required?: boolean` and create our own option that uses this `required` argument when we call `core.getInput` below

Reply:
Updated.